### PR TITLE
Creates student reviews pages and college info page styling.

### PR DESCRIPTION
**Pull request by Sydney-18**

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -10,4 +10,85 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Query query = new Query(Comment.MESSAGE_KEY).addSort(Comment.TIMESTAMP_KEY, SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<Comment>(); 
+
+    // limit 10 comments
```
Comment (tobaolo) [inline src/main/java/com/google/sps/servlets/DataServlet.java:52]: Capitalize L in limit and add a period at the end.

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -10,4 +10,85 @@
...
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/data")
+public class DataServlet extends HttpServlet {
+
+  private static final SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Query query = new Query(Comment.MESSAGE_KEY).addSort(Comment.TIMESTAMP_KEY, SortDirection.DESCENDING);
```
Comment (tobaolo) [inline src/main/java/com/google/sps/servlets/DataServlet.java:45]: Delete the line between the doGet method head and the first line in the function (Query query ...)

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -10,4 +10,85 @@
...
+      long timestamp = (long) entity.getProperty(Comment.TIMESTAMP_KEY);
+      String time = (String) entity.getProperty(Comment.TIME_KEY);
+
+      Comment comment = new Comment(name, message, timestamp, time);
+      comments.add(comment); 
+    }
+
+    final Gson gson = new Gson();
+    String jsonComments = gson.toJson(comments);
+    response.setContentType("text/html;");
```
Comment (qi-google) [inline src/main/java/com/google/sps/servlets/DataServlet.java:65]: a drive-by comment: shouldn't the http header be `application/json`?

File: src/main/webapp/comments.html [hunk 4]
```diff
@@ -5,6 +5,25 @@
...
     <script src="script.js"></script>
   </head>
-  <body onload="getSchoolInfo()">
+  <body>
+    <div id="content">
+      <div class="header">
+        <h1>Insightful</h1>
+      </div>   
+      <div id="form-div">
+        <a href="index.html">< Back to Profile Home</a>
```
Comment (qi-google) [inline src/main/webapp/comments.html:14]: Remove the extra `<` tag?

**Comment by tobaolo** [comment]
@googlebot I signed it!

**Comment by Sydney-18** [comment]
@googlebot I signed it!

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -10,4 +10,83 @@
...
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query(Comment.MESSAGE_KEY).addSort(Comment.TIMESTAMP_KEY, SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<Comment>(); 
+
+    for(Entity entity : results.asIterable(FetchOptions.Builder.withLimit(10))) {
```
Comment (qi-google) [inline src/main/java/com/google/sps/servlets/DataServlet.java:51]: is this on purpose? So basically this will only fetch the first 10 comments right? You may want to add comments indicating that this is for MVP for now.

Nit: I would extract `results.asIterable(FetchOptions.Builder.withLimit(10))` into a variable like 
```
PreparedQuery preparedQuery = datastore.prepare(query);
// For MVP, only fetch the first 10 comments for now.
Iterable<Entity> entities = preparedQuery.asIterable(FetchOptions.Builder.withLimit(10));
for (Entity entiry : entities) {
``` 

to make the code more clear.

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -10,4 +10,83 @@
...
+    response.getWriter().println(jsonComments);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = getParameter(request, "name-input", "");
+    String message = getParameter(request, "text-input", "");
+    long timestamp = System.currentTimeMillis();
+
+    Date date = new Date(timestamp);
```
Comment (qi-google) [inline src/main/java/com/google/sps/servlets/DataServlet.java:73]: `Date date = new Date();` should be sufficient without the timestamp here?

File: src/main/webapp/college-info.html [hunk 7]
```diff
@@ -6,15 +6,32 @@
...
-    <input type="text" id="school-search"> 
-    <button onclick="getSchoolInfo()">Search</button>
-    <h1 id="school-name"></h1>
-    <h3 id="school-desc"></h3>
-    <div id="students"></div>
-    <div id="race-piechart" style="width: 900px; height: 500px;"></div>
-    <div id="gender-piechart" style="width: 900px; height: 500px;"></div>
-    <div id="admissions"></div>
-    <div id="cost"></div>
+  <body onload="getSchoolInfo()">
```
Comment (qi-google) [inline src/main/webapp/college-info.html:9]: mm, I remembered that `onload=getSchoolInfo()` doesn't work right? @tobaolo Shouldn't we keep the button with `onlick` event here?

File: src/main/webapp/style.css [hunk 8]
```diff
@@ -0,0 +1,273 @@
+body {
+background-color: #ECECEC;
```
Comment (qi-google) [inline src/main/webapp/style.css:2]: Nit: indent two spaces?

File: src/main/webapp/college-info.html (hunk 7 above)
Comment (tobaolo) [reply src/main/webapp/college-info.html:9]: ahh yes. this got messed up in the merge. i will fix that thanks

**Comment by tobaolo** [comment]
@googlebot I signed it!

**Comment by tobaolo** [comment]
@googlebot I signed it!

File: src/main/webapp/script.js [hunk 9]
```diff
@@ -216,3 +216,24 @@ function getLink(schoolName) {
       'school.main_campus,school.institutional_characteristics.level')
 }
 
+// Adds comments to page
```
Comment (scottymb) [inline src/main/webapp/script.js:219]: Format this as a JsDoc comment:

/** Adds comments to page. */

File: src/main/webapp/script.js [hunk 10]
```diff
@@ -216,3 +216,24 @@ function getLink(schoolName) {
...
+function loadComments() {
+  fetch('/data').then(response => response.json()).then((comments) => {
+    const commentListElement = document.getElementById('comments-container');
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment.name,
+        comment.message, comment.time));
```
Comment (scottymb) [inline src/main/webapp/script.js:225]: Indent another 2 spaces (4 spaces total for continued long lines).

File: src/main/webapp/script.js [hunk 11]
```diff
@@ -216,3 +216,24 @@ function getLink(schoolName) {
...
+    });
+  });
+}
+
+/**
+ * @param {string} name The name of the user who commented.
+ * @param {string} message The message body of a comment post.
+ * @param {string} time The time of a comment post.
+ */
```
Comment (scottymb) [inline src/main/webapp/script.js:234]: Add an @return {!HTMLLIElement} for the element created.

**Comment by Sydney-18** [comment]
@googlebot I signed it!

File: src/main/webapp/comments.html [hunk 12]
```diff
@@ -5,6 +5,25 @@
...
+      </div>   
+      <div id="form-div">
+        <a href="college-info.html">Back to Profile Home</a>  
+        <div id="comments-header">Student Reviews</div>
+        <div id="comments-container">
+          <h5>Student • June 2, 2020</h5>
+          <p>I love this school and recommend it 100%! The professors are caring and the administration is as well.</p>
+        </div>
+        <form action="/data" method="POST">
+          <div id="comments-header">Leave a review:</div>
```
[the remark this answers is not in the document]
Comment (Sydney-18) [reply src/main/webapp/comments.html:21]: Would line 15 have to be 'p' as well?

[the remark this answers is not in the document]
Comment (scottymb) [reply src/main/webapp/comments.html:21]: Following up on this after talking in person - we decided this should move outside of the form and use the same `hX` tag used for line 15.

File: src/main/webapp/style.css [hunk 13]
```diff
@@ -0,0 +1,273 @@
...
+  font-weight: normal;
+  height: 58.54px;
+  line-height: 86px; 
+  margin: auto;
+  margin-bottom: 15px;
+  text-align: center;
+  width: 1035px;
+}
+
+button {
```
[the remark this answers is not in the document]
Comment (Sydney-18) [reply src/main/webapp/style.css:107]: How would I start that?

[the remark this answers is not in the document]
Comment (scottymb) [reply src/main/webapp/style.css:107]: It's just about moving around these blocks. Move all of the selectors that are tag names rather than classes or ids to the top of the file. You'd have:

body {
...
}

p {...}

a {...}

button {...}

input {...}

input:hover {...}

textarea {...}

etc.

File: src/main/webapp/college-info.html [hunk 14]
```diff
@@ -7,14 +7,31 @@
...
-    <div id="gender-piechart" style="width: 900px; height: 500px;"></div>
-    <div id="admissions"></div>
-    <div id="cost"></div>
+    <header>
+      <h1>Black Student Insight</h1>
+      <input type="text" id="school-search"> 
+      <search-button onclick="getSchoolInfo()">Search</search-button> 
+    </header>
+    <main>
+      <div class="school-header">
```
Comment (scottymb) [inline src/main/webapp/college-info.html:16]: I believe you can handle all of the styling on this header without this div. Correct me if I'm wrong, but if you just removed it and moved all of the .school-header styles into #school-name, I think you'd be pretty good to go.

File: src/main/webapp/college-info.html [hunk 15]
```diff
@@ -7,14 +7,31 @@
...
+      <section>  
+        <p id="race-piechart"></p>
+        <p id="gender-piechart"></p>
+      </section>
+      <a class="reviews-button" href="comments.html">Student Reviews ></a>
+    </main>  
   </body>
-</html>
\ No newline at end of file
+</html>
```
Comment (scottymb) [inline src/main/webapp/college-info.html:37]: This is all really great, so much cleaner!

File: src/main/webapp/comments.html [hunk 16]
```diff
@@ -5,6 +5,23 @@
     <link rel="stylesheet" href="style.css">
     <script src="script.js"></script>
   </head>
-  <body onload="getSchoolInfo()">
+  <body>
+    <div id="content">
+      <header><h1>Black Student Insight</h1></header>   
+      <div>
```
Comment (scottymb) [inline src/main/webapp/comments.html:11]: `main` tag here perhaps, instead of `div`?

File: src/main/webapp/comments.html [hunk 17]
```diff
@@ -5,6 +5,23 @@
...
+        <a href="college-info.html">Back to Profile Home</a>  
+        <h2>Student Reviews</h2>
+        <div id="comments-container">
+          <h5>Student • June 2, 2020</h5>
```
Comment (scottymb) [inline src/main/webapp/comments.html:15]: Just a little note on this one, again. h3 or h4 instead?

File: src/main/webapp/script.js [hunk 18]
```diff
@@ -216,3 +216,25 @@ function getLink(schoolName) {
...
+          comment.message, comment.time));
+    });
+  });
+}
+
+/**
+ * @param {string} name The name of the user who commented.
+ * @param {string} message The message body of a comment post.
+ * @param {string} time The time of a comment post.
+   @return {HTMLLIElement} commentElement.
```
Comment (scottymb) [inline src/main/webapp/script.js:234]: Start the line with a "*".

Comment (scottymb) [reply src/main/webapp/script.js:234]: Also, while @param needs the param name, @return just takes an optional description of the returned element. For you, maybe it's `@return {!HTMLLIElement} A comment list item. Be sure to add the "!" which shows that it can't be null. This is a bit confusing - it's not needed for all primitive JavaScript types, which are `string`, `number, and `boolean`, since they're nonnullable by default. Everything else must have ! for nonnullable or ? for nullable.

File: src/main/webapp/script.js [hunk 19]
```diff
@@ -216,3 +216,25 @@ function getLink(schoolName) {
       'school.main_campus,school.institutional_characteristics.level')
 }
 
+/** Adds comments to page. */
+function loadComments() {
+  fetch('/data').then(response => response.json()).then((comments) => {
+    const commentListElement = document.getElementById('comments-container');
```
Comment (scottymb) [inline src/main/webapp/script.js:222]: I just realized - `li` tags are only allowed to be children of `ul` or `ol` elements, not `div`. I think your easiest bet here is to create a `p` tag instead of list item.

File: src/main/webapp/style.css [hunk 20]
```diff
@@ -0,0 +1,236 @@
...
+  font-family: Aksara Bali Galang;
+}
+
+main {
+  margin: auto;
+  max-width: 1035px;
+  width: 100%;  
+}
+
+#content {
```
Comment (scottymb) [inline src/main/webapp/style.css:13]: Is this layer and these style necessary? It looks to me like this will essentially do the same thing as if it weren't there, but correct me if I'm missing anything.

File: src/main/webapp/style.css [hunk 21]
```diff
@@ -0,0 +1,236 @@
...
+  color: #787878;  
+  display: block;
+  font-size: 18px;
+  font-style: normal;
+  font-weight: normal;
+  height: 26px;
+  line-height: 49px;
+  margin: auto;
+  margin-bottom: 15px;
+  width: 955px;
```
Comment (scottymb) [inline src/main/webapp/style.css:43]: Can this just rely on its container's width instead of defining its own?

File: src/main/webapp/style.css [hunk 22]
```diff
@@ -0,0 +1,236 @@
...
+  text-align: center;
+}
+
+h5 {
+  color: #787878;  
+  display: block;
+  font-size: 18px;
+  font-style: normal;
+  font-weight: normal;
+  height: 26px;
```
Comment (scottymb) [inline src/main/webapp/style.css:39]: Can you remove this and address with line-height, margin, and/or padding instead?

File: src/main/webapp/style.css [hunk 23]
```diff
@@ -0,0 +1,236 @@
...
+  background: #856c8b;
+  border: 1px solid black;
+  box-shadow: 0 4px 4px #00000040;
+  box-sizing: border-box;
+}
+
+#school-search {
+  float: right; 
+  height: 35px; 
+  position: absolute;
```
Comment (scottymb) [inline src/main/webapp/style.css:114]: I wouldn't suggest using both float and position absolute together. Floating and using right and top margins should work for you, but if this is giving you troubles we can defer for now and look again later.

File: src/main/webapp/style.css [hunk 24]
```diff
@@ -0,0 +1,236 @@
...
+
+#school-name {
+  color: #856c8b; 
+  font-size: 30px; 
+  margin: auto;
+  padding: 22px;
+  text-align: center;
+}
+
+#school-desc {
```
Comment (scottymb) [inline src/main/webapp/style.css:158]: I'm noticing that these sections have a lot of repeated styles. I recommend grouping all of the shared styles together in one rule and specify all of the ID selectors they apply to. Then separately break out each one like you've done, but only add the unique styles.

File: src/main/webapp/style.css [hunk 25]
```diff
@@ -0,0 +1,236 @@
...
+  background: white;
+  box-shadow: 0 4px 4px #00000040;  
+  float: left;
+  height: 50px;
+  margin-top: 20px;
+  padding: 25px; 
+  top: 290px;
```
Comment (scottymb) [inline src/main/webapp/style.css:165]: These `top` values should be doing nothing since you aren't setting a specific `position` value. Can you check and remove assuming possible?

File: src/main/webapp/script.js [hunk 26]
```diff
@@ -231,10 +231,10 @@ function loadComments() {
  * @param {string} name The name of the user who commented.
  * @param {string} message The message body of a comment post.
  * @param {string} time The time of a comment post.
-   @return {HTMLLIElement} commentElement.
+ * @return {!HTMLLIElement} A comment list item.
```
Comment (scottymb) [inline src/main/webapp/script.js:234]: This now changes to {!HTMLParagraphElement}
